src/lib/node.ts

```diff
@@ -230,13 +231,25 @@ export function pipeToNodeStream<T>(
  * @param nodeReadable Soource NodeJS stream
  * @returns Readable stream of type `string | Buffer`
  */
-export function fromNodeReadable(
+export function fromNodeReadable<T = void>(
 	nodeReadable: NodeJS.ReadableStream
-): ReadableStream<string | Buffer> {
-	const stream = new Stream<string | Buffer>();
+): ReadableStream<FromNodeReadableDataType<T>> {
+	const stream = new Stream<FromNodeReadableDataType<T>>();
 
 	const endStream = async (err?: Error) => {
-		if (!stream.isEndingOrEnded()) return stream.end(err);
+		if (!stream.isEndingOrEnded()) {
+			/**
+			 * `stream.end` could reject, so we'll emit
+			 * any caught errors as an event via the NodeJS stream.
+			 * This is acceptable as we are only attempting to end
+			 * the ts-stream a maximum one time.
+			 */
+			try {
+				return stream.end(err);
```

Comment:
Can you add a test for this?
Because I'm pretty sure it doesn't work right now (as the promise is directly returned, a rejection will not be caught).

Reply:
Could you help me figure out how to test this actually? Under what conditions would `stream.end` reject? I could just override the method to force it to reject, but then the stream would never finish, right?

```ts
const stream = new Stream<string>();
stream.end = () => Promise.reject("uh oh!");

// This will just hang because the ts-stream cannot properly end... 🤔 
const result = await fromNodeReadable(someReadable).toArray();
```

Reply:
Please see my remark at your latest tests. You'll need to use `.forEach()`, and instead of returning a write failure (first argument), just return a failure to the ender callback (second argument).